### PR TITLE
refactored podium worker to permit its use on another application

### PR DESCRIPTION
**Pull request by lucas-machado**
Made it possible to use the worker code as lib from another project.

### CODE REVIEW

File: worker/expiration_worker.go [hunk 1]
```diff
@@ -238,22 +233,27 @@ func (w *ExpirationWorker) Run() {
 			}
 			if w.running == false {
 				w.running = true
-				result, err := w.expireScores()
+				results, err := w.expireScores()
 				if err != nil {
-					l.Error("error expiring scores", zap.Error(err))
+					errChan <- fmt.Errorf("error expiring scores: %v", err)
+					break
```
Comment (cscatolini) [inline worker/expiration_worker.go]: Are you sure we should stop the worker if any error occurs? I think we should exit if we reach some threshold (e.g. if redis is down) but not on the first error.

File: worker/expiration_worker.go [hunk 2]
```diff
@@ -44,37 +42,60 @@ type ExpirationWorker struct {
...
+		return nil, err
+	}
+
 	return worker, nil
 }
 
-func (w *ExpirationWorker) configure() error {
-	err := w.loadConfiguration()
+// GetExpirationWorkerExtCfg returns a new scores expirer worker with already loaded configuration.
+func GetExpirationWorkerExtCfg(host string, port int, password string, db int, connectionTimeout int,
```
Comment (cscatolini) [inline worker/expiration_worker.go]: I think `NewExpirationWorker` would be a better name.

Comment (lucas-machado) [reply worker/expiration_worker.go]: Makes sense. Changing it.

File: worker/expiration_worker.go (hunk 1 above)
Comment (lucas-machado) [reply worker/expiration_worker.go]: Reading about redis internally it seems that they keep trying to execute operations specifically when there are network errors, eventually sleeping if there are enough errors. Restarting processing if the server goes back online. So I think the best way to solve this, for now, is to use an error channel merely for reporting and keep the worker running.

Comment (lucas-machado) [reply worker/expiration_worker.go]: @cscatolini PTAL

**Pull request merged**